### PR TITLE
PLT-929:Fixed Virtual cluster day 2 operation

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/cluster_common_fields.go [hunk 1]
```diff
@@ -23,10 +23,12 @@ func readCommonFields(c *client.V1Client, d *schema.ResourceData, cluster *model
 		return diag.FromErr(err), true
 	}
 	// When the current repave state is pending, we set the review_repave_state to Pending, For indicate the system change.
-	if err := d.Set("review_repave_state", cluster.Status.Repave.State); err != nil {
-		return diag.FromErr(err), true
+	if _, ok := d.GetOk("review_repave_state"); ok {
+		// We are adding this check to handle virtual cluster scenario. virtual cluster doesn't have support for `review_repave_state`
+		if err := d.Set("review_repave_state", cluster.Status.Repave.State); err != nil {
```
Comment (nikchern) [inline spectrocloud/cluster_common_fields.go:28]: @SivaanandM could you please add a unit test scenario to cover this branch?

Comment (SivaanandM) [reply spectrocloud/cluster_common_fields.go:28]: its already added as part of cluster

**Review by nikchern — CHANGES_REQUESTED** [review]
@SivaanandM in fact we need to improve unit test coverage for readCommonFields()

**Comment by SivaanandM** [comment]
> @SivaanandM in fact we need to improve unit test coverage for readCommonFields()

added one with vsphere and virtual cluster

**Review by nikchern — APPROVED** [review]

**Pull request merged**